### ISSUE TITLE
"Global" pipelines

### ISSUE THREAD

**Issue by qwerty287** [woodpecker-ci/woodpecker#1070]
Add an option to define "global" pipelines that are added to *every* pipeline of the user the global pipeline belongs to. My main usecase would be failing notifications (also see https://codeberg.org/Codeberg-CI/feedback/issues/5) that are added to every pipeline so I don't have to add them manually. This would require some more dicussion: should these pipelines be visible to every user or only to the one it belongs to? How to define global pipelines, using a special repo on the forge or an internal editor?

**Comment by lafriks** [comment]
Also would be cool if it would be possible to define pipeline or multiple pipelines for repository that would be run even if repository itself does not have pipeline definition. At least in my company we have quite many pipelines that are practically the same for many repositories and this would greatly reduce burden to maintain them.

One other nice thing would be if you could define variables (like secrets but specific for that pipeline) that could be set in repository settings and than passed to pipeline as variables.

**Comment by anbraten** [comment]
linking #1089 (maybe a duplicate)

**Comment by anbraten** [comment]
> One other nice thing would be if you could define variables (like secrets but specific for that pipeline) that could be set in repository settings and than passed to pipeline as variables.

Yeah I had that in mind as well. My idea was to change secrets to masked environment variables and by that allow to have unmasked variables as well.

**Comment by EchedeyLR** [comment]
> One other nice thing would be if you could define variables (like secrets but specific for that pipeline) that could be set in repository settings and than passed to pipeline as variables.

@lafriks I guess this would solve the use of the same environment variables in the same repository too. By having a main and global pipeline either at repository level or organization level where variables are taken.

In my case, I need it at repository level to build different architectures in different pipelines.

**Comment by anbraten** [comment]
looking at this again, I think this is some custom logic which could be done by using the external configuration service #783 and simply passing the pipeline config through, but adding a fail step to each workflow.
